packages/backend-function/src/factory.test.ts

```diff
@@ -337,5 +337,15 @@ void describe('AmplifyFunctionFactory', () => {
         new Error('runtime must be one of the following: 16, 18, 20')
       );
     });
+
+    void it('throws when the oldest maintained Node LTS reaches end of life', () => {
+      // Date when the oldest Node LTS maintenance ends according to https://github.com/nodejs/release#release-schedule.
+      // Once this test fails, update endDate to the end date of the next Node LTS version.
+      // After updating the date, next would be updating the default function runtime in factory.ts
+      const endDate = new Date('2025-04-30');
```

Comment:
should this test fail ~month before or so? so we get a chance to plan and execute remediation ?

Reply:
Yeah we can do that, a month should be enough time in case anything crazy happens during release but customer's will have the updated runtime before the old runtime maintenance ends.